### PR TITLE
Use doc.path() instead of doc.relative_path() for the buffer picker

### LINKED ISSUE #619: Helix panics under certain conditions
tested with commit: https://github.com/helix-editor/helix/commit/098b6b6eedf6da8e760b334d61acef56be8d258b

**steps to reproduce:**
- cargo new project
- move cursor to println
- `gd` (go to definition)
- (space b) to open buffers
- click down arrow repeatedly

**backtrace:**
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Line index out of bounds: line index 79, Rope/RopeSlice line count 2', /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/ropey-1.3.1/src/slice.rs:338:41
stack backtrace:
   0:     0x55ba3f38b400 - std::backtrace_rs::backtrace::libunwind::trace::h34055254b57d8e79
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55ba3f38b400 - std::backtrace_rs::backtrace::trace_unsynchronized::h8f1e3fbd9afff6ec
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55ba3f38b400 - std::sys_common::backtrace::_print_fmt::h3a99a796b770c360
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55ba3f38b400 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h32d1f94a80615d18
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55ba3f3b12bc - core::fmt::write::h306731c068f7162c
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/fmt/mod.rs:1110:17
   5:     0x55ba3f385ea5 - std::io::Write::write_fmt::hd2fa90334eee2a21
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/io/mod.rs:1588:15
   6:     0x55ba3f38d71b - std::sys_common::backtrace::_print::h5abaa2601a852287
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55ba3f38d71b - std::sys_common::backtrace::print::h8d81445442bb638f
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55ba3f38d71b - std::panicking::default_hook::{{closure}}::hcfe804496a9fa747
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:208:50
   9:     0x55ba3f38d1f1 - std::panicking::default_hook::hbea8e3ccf2ba8901
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:225:9
  10:     0x55ba3e516e43 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h9bf159e76ed6947d
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/alloc/src/boxed.rs:1589:9
  11:     0x55ba3e5210a3 - helix_term::application::Application::run::{{closure}}::{{closure}}::h48924ddc35f5e087
                               at /workspace/helix/helix-term/src/application.rs:525:13
  12:     0x55ba3f38de09 - std::panicking::rust_panic_with_hook::h7ee9e1a2d0f8975a
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:626:17
  13:     0x55ba3f38d8c7 - std::panicking::begin_panic_handler::{{closure}}::h8ab3b4491718b2c7
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:519:13
  14:     0x55ba3f38b8fc - std::sys_common::backtrace::__rust_end_short_backtrace::hd489062ffa586a9f
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:141:18
  15:     0x55ba3f38d829 - rust_begin_unwind
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:515:5
  16:     0x55ba3e4a89f1 - core::panicking::panic_fmt::hca6330e3e14086b4
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/panicking.rs:92:14
  17:     0x55ba3e4a8ae3 - core::result::unwrap_failed::h6160d2b4bfbb0e87
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/result.rs:1355:5
  18:     0x55ba3f35e083 - core::result::Result<T,E>::unwrap::h90913094869577c6
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/result.rs:1037:23
  19:     0x55ba3e743ec0 - ropey::slice::RopeSlice::line_to_byte::h1f24e32cd20e3c10
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/ropey-1.3.1/src/slice.rs:338:9
  20:     0x55ba3e77ffe0 - helix_term::ui::editor::EditorView::doc_syntax_highlights::he7436d313978d284
                               at /workspace/helix/helix-term/src/ui/editor.rs:148:25
  21:     0x55ba3e809b26 - <helix_term::ui::picker::FilePicker<T> as helix_term::compositor::Component>::render::hda1a42f7c80b2021
                               at /workspace/helix/helix-term/src/ui/picker.rs:123:30
  22:     0x55ba3e7d15f6 - helix_term::compositor::Compositor::render::h8709a94782894ec6
                               at /workspace/helix/helix-term/src/compositor.rs:156:13
  23:     0x55ba3e7c5080 - helix_term::application::Application::render::h67d851cc06e9bee8
                               at /workspace/helix/helix-term/src/application.rs:151:9
  24:     0x55ba3e7c53b1 - helix_term::application::Application::handle_terminal_events::h7ce643d170ad878e
                               at /workspace/helix/helix-term/src/application.rs:244:13
  25:     0x55ba3e51aadb - helix_term::application::Application::event_loop::{{closure}}::hcef07f53f870225d
                               at /workspace/helix/helix-term/src/application.rs:173:21
  26:     0x55ba3e4ecc5b - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hbb1e8992dc5b7dc1
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/future/mod.rs:80:19
  27:     0x55ba3e520ab7 - helix_term::application::Application::run::{{closure}}::h696ef32bf1411f45
                               at /workspace/helix/helix-term/src/application.rs:528:9
  28:     0x55ba3e4ece89 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hf80daa9f852a215d
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/future/mod.rs:80:19
  29:     0x55ba3e4e5bc6 - hx::main::{{closure}}::h35c3a36a0bb2922d
                               at /workspace/helix/helix-term/src/main.rs:97:5
  30:     0x55ba3e4eca09 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::ha7d62983e0b29a95
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/future/mod.rs:80:19
  31:     0x55ba3e519250 - tokio::park::thread::CachedParkThread::block_on::{{closure}}::hd8022e584cf11138
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/park/thread.rs:263:54
  32:     0x55ba3e4fd572 - tokio::coop::with_budget::{{closure}}::h6887b42ccdd6e06d
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/coop.rs:106:9
  33:     0x55ba3e4fe683 - std::thread::local::LocalKey<T>::try_with::h55d748879f156425
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/thread/local.rs:400:16
  34:     0x55ba3e4fe28d - std::thread::local::LocalKey<T>::with::h1027de06688f7005
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/thread/local.rs:376:9
  35:     0x55ba3e518b81 - tokio::coop::with_budget::hd5d86e236da521ef
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/coop.rs:99:5
  36:     0x55ba3e518b81 - tokio::coop::budget::h464bff956f0c34bb
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/coop.rs:76:5
  37:     0x55ba3e518b81 - tokio::park::thread::CachedParkThread::block_on::h6d6b602d6944112f
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/park/thread.rs:263:31
  38:     0x55ba3e51933e - tokio::runtime::enter::Enter::block_on::h8ec004776dbbcb95
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/runtime/enter.rs:151:13
  39:     0x55ba3e5017c3 - tokio::runtime::thread_pool::ThreadPool::block_on::ha46c36d7a6ca1a50
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/runtime/thread_pool/mod.rs:71:9
  40:     0x55ba3e4ec395 - tokio::runtime::Runtime::block_on::h8aef2bf64157ac8b
                               at /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.10.0/src/runtime/mod.rs:452:43
  41:     0x55ba3e5016c7 - hx::main::h757ba648e423133c
                               at /workspace/helix/helix-term/src/main.rs:99:5
  42:     0x55ba3e5093eb - core::ops::function::FnOnce::call_once::h84dabc8ac691cb91
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/ops/function.rs:227:5
  43:     0x55ba3e5108be - std::sys_common::backtrace::__rust_begin_short_backtrace::hfaa317d7a7eca987
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/sys_common/backtrace.rs:125:18
  44:     0x55ba3e503041 - std::rt::lang_start::{{closure}}::h092172ae08fe73e0
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/rt.rs:49:18
  45:     0x55ba3f38e3e9 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h2aabc384aab89b7b
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/ops/function.rs:259:13
  46:     0x55ba3f38e3e9 - std::panicking::try::do_call::hc5fcacb7a85fc7b1
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:401:40
  47:     0x55ba3f38e3e9 - std::panicking::try::hb5d9603af3abbe3a
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:365:19
  48:     0x55ba3f38e3e9 - std::panic::catch_unwind::h98fe6ac3925e64b4
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panic.rs:434:14
  49:     0x55ba3f38e3e9 - std::rt::lang_start_internal::h22ac7383c516f93e
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/rt.rs:34:21
  50:     0x55ba3e503020 - std::rt::lang_start::h3a4616dbbd680322
                               at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/rt.rs:48:5
  51:     0x55ba3e50174c - main
  52:     0x7f207b808b25 - __libc_start_main
  53:     0x55ba3e4a91ee - _start
  54:                0x0 - <unknown>
```

**demo**

https://user-images.githubusercontent.com/22427111/129952264-e7836d62-cd40-4929-9692-95db1459b898.mp4

### PR DESCRIPTION
**Pull request by kirawi**
Resolves #619

Not sure if `buffers_remaining_impl` also needs to be similarly changed.

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -2240,7 +2240,7 @@ fn buffer_picker(cx: &mut Context) {
         cx.editor
             .documents
             .iter()
-            .map(|(id, doc)| (id, doc.relative_path()))
```
Comment (archseer) [inline helix-term/src/commands.rs:2243]: doc.path().cloned()

**Review by archseer — CHANGES_REQUESTED** [review]
format_fn should still display path as relative

**Comment by kirawi** [comment]
Ah, I'll get it ready to merge tomorrow. Forgot one thing.

**Comment by archseer** [comment]
I wrote an alternative in https://github.com/helix-editor/helix/commit/1d45f50781f34bccf29a3e4f576cc48651038b09

**Pull request closed without merging**